File: client/src/app/test/constants/fake-bracket.ts
/* eslint-disable @typescript-eslint/no-magic-numbers -- Pour définir plusieurs utilisateurs*/
import { BracketUser, PlayerGameState } from '@common/interfaces/tournament/bracket-profile';
import { CommonBracket } from '@common/interfaces/tournament/common-bracket';

const FAKE_USER: (id: number, winner: PlayerGameState) => BracketUser = (id: number, winner: PlayerGameState) => {
    return {
        id: `${id}`,
        name: `User mmmmmmm ${id}`,
        winner,
        profilePicture: '',
    };
};

const FAKE_BRACKET_NODE_1: () => CommonBracket = () => {
    return {
        children: [],
        currentPlayers: [FAKE_USER(1, PlayerGameState.WINNER), FAKE_USER(2, PlayerGameState.LOSER)],
        isMatchInProgress: false,
        gameId: '1',
    };
};

const FAKE_BRACKET_NODE_2: () => CommonBracket = () => {
    return {
        children: [],
        currentPlayers: [FAKE_USER(3, PlayerGameState.NO_RESULT), FAKE_USER(4, PlayerGameState.LOSER)],
        isMatchInProgress: true,
        gameId: '2',
    };
};

export const FAKE_BRACKET_1: () => CommonBracket = () => {
    return {
        children: [FAKE_BRACKET_NODE_1(), FAKE_BRACKET_NODE_2()],
        currentPlayers: [FAKE_USER(1, PlayerGameState.NO_RESULT)],
        isMatchInProgress: false,
        gameId: '3',
    };
};

export const FAKE_CONSOLATION_BRACKET_1: () => CommonBracket = () => {
    return {
        children: [],
        currentPlayers: [FAKE_USER(2, PlayerGameState.NO_RESULT), FAKE_USER(5, PlayerGameState.NO_RESULT)],
        isMatchInProgress: true,
        gameId: '4',
    };
};

File: mobile/lib/constants/fake/fake_bracket.dart
import 'package:Scrabble/data/models/enums/bracket_result.dart';
import 'package:Scrabble/data/models/interfaces/bracket/bracket_user.dart';
import 'package:Scrabble/data/models/interfaces/bracket/common_bracket.dart';

BracketUser FAKE_BRACKET_USER(int id, int winner) {
  return BracketUser(id.toString(), 'User mmmmmmm $id', winner, '');
}

CommonBracket FAKE_BRACKET_NODE_1() {
  return CommonBracket([], [
    FAKE_BRACKET_USER(1, bracketResultToInt(BracketResult.WINNER)),
    FAKE_BRACKET_USER(2, bracketResultToInt(BracketResult.LOSER))
  ], false, '');
}

CommonBracket FAKE_BRACKET_NODE_2() {
  return CommonBracket([], [
    FAKE_BRACKET_USER(3, bracketResultToInt(BracketResult.NO_RESULT)),
    FAKE_BRACKET_USER(4, bracketResultToInt(BracketResult.LOSER))
  ], true, '');
}

CommonBracket FAKE_BRACKET_1() {
  return CommonBracket([FAKE_BRACKET_NODE_1(), FAKE_BRACKET_NODE_2()],
      [FAKE_BRACKET_USER(1, bracketResultToInt(BracketResult.NO_RESULT))], false, '');
}

CommonBracket FAKE_CONSOLATION_BRACKET_1() {
  return CommonBracket([], [
    FAKE_BRACKET_USER(2, bracketResultToInt(BracketResult.NO_RESULT)),
    FAKE_BRACKET_USER(5, bracketResultToInt(BracketResult.NO_RESULT))
  ], true, '');
}
